imports/wasi_snapshot_preview1/poll.go

```diff
@@ -161,44 +166,34 @@ func pollOneoffFn(_ context.Context, mod api.Module, params []uint64) sys.Errno
 		}
 	}
 
-	// If there are subscribers with data ready, we have already written them to outBuf,
-	// and we don't need to wait for the timeout: clear it.
-	if readySubs != 0 {
-		timeout = 0
-	}
+	sysCtx := mod.(*wasm.ModuleInstance).Sys
 
-	// If there are blocking stdin subscribers, check for data with given timeout.
-	if len(blockingStdinSubs) > 0 {
-		stdin, ok := fsc.LookupFile(internalsys.FdStdin)
-		if !ok {
-			return sys.EBADF
-		}
-		// Wait for the timeout to expire, or for some data to become available on Stdin.
-		stdinReady, errno := stdin.File.Poll(sys.POLLIN, int32(timeout.Milliseconds()))
+	// There are no blocking subscribers, we just wait for the given timeout.
```

Comment:
I know this is old logic, but is the intent to wait for the timeout? is the assumption only clock events left, or does this loop? Or does wasi-libc expect the same? Might be worth clarifying why an unconditional timeout?

Reply:
yes the assumption is that only clock events are left, the timeout is initialized earlier at lines ~122

https://github.com/tetratelabs/wazero/blob/1f8c908f1c61c964192c731af495caf1d720418c/imports/wasi_snapshot_preview1/poll.go#L123-L126

I will update the comment to be more explicit about it.

Reply:
I have now refactored all the logic and added a better explanation